apps/api/pages/api/event-types/by-slug/[slug]/_auth-middleware.ts

```diff
@@ -0,0 +1,33 @@
+import type { NextApiRequest } from "next";
+
+import { HttpError } from "@calcom/lib/http-error";
+
+import { schemaQuerySlug } from "~/lib/validations/shared/querySlug";
+
+async function authMiddleware(req: NextApiRequest) {
+  const { userId, isAdmin, prisma } = req;
+  const { slug, ownerId } = schemaQuerySlug.parse(req.query);
+
+  const effectiveUserId = isAdmin && ownerId ? ownerId : userId;
+
+  const eventType = await prisma.eventType.findFirst({
```

Comment:
This will always return the first `event-type` with the given `slug` from the DB. So this will always throw forbidden for all the users except admin and the owner of the first `event-type` with the given slug.

Reply:
Thank you for your fast review, appreciate and well spotted! Pushed update in commit [fix: auth middleware](https://github.com/calcom/cal.com/pull/13374/commits/3123ae4b6249995668c3911b87017274eb3b9751).

Reply:
Alternatively, I was thinking we could `.findUnique` by `EventType.slug` and `EventType.userId` 🤔

Reply:
No, the `.findUnique` expects at least one unique field. I think `.findFirst` is good here.